triton/stream_reader.go

```diff
@@ -37,13 +37,19 @@ func (msr *multiShardStreamReader) Checkpoint() (err error) {
 	return
```

Comment:
Seems like if you're going to have the shard sequence number state managed outside the ShardStream itself, maybe it should ONLY be managed/tracked there?

Seems like it'll be confusing if there are two places the sequence numbers are recorded right?